src/credentials/DeviceAttestationConstructor.h

```diff
@@ -0,0 +1,65 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
```

Comment:
Note to other reviewers on this PR: all files from `src/credentials/` on this PR are currently being worked in https://github.com/project-chip/connectedhomeip/pull/9544, but since this PR depends on the other one, they are manually synced here. If comments arise re: `src/credentials`, consider heading over to #9544

Reply:
merged